x-pack/test/detection_engine_api_integration/utils.ts

```diff
@@ -1324,15 +1325,22 @@ export const waitForRuleSuccessOrStatus = async (
             )}, status: ${JSON.stringify(response.status)}`
           );
         }
-        if (response.body[id]?.current_status?.status !== status) {
+        const currentStatus = response.body[id]?.current_status;
+
+        if (currentStatus?.status !== status) {
           log.debug(
             `Did not get an expected status of ${status} while waiting for a rule success or status for rule id ${id} (waitForRuleSuccessOrStatus). Will continue retrying until status is found. body: ${JSON.stringify(
               response.body
             )}, status: ${JSON.stringify(response.status)}`
           );
         }
-
-        return response.body[id]?.current_status?.status === status;
+        if (currentStatus == null) {
```

Comment:
Nit: IMO this is cleaner with either `return currentStatus != null && currentStatus.status === status ...` or `return currentStatus?.status === status && ...`.